code/03_census_and_bed_days.R

```diff
@@ -0,0 +1,868 @@
+
+labedstemplate <- labedstemplate %>% 
+  rename(LocalAuthorityArea = LA,
+         age_grp = age)
+
+labedstemplate<-labedstemplate%>% mutate(LocalAuthorityArea=toupper(LocalAuthorityArea),age_grp=toupper(age_grp),reas1=toupper(reas1))
+datafile20<-datafile20%>% mutate(LocalAuthorityArea=toupper(LocalAuthorityArea),age_grp=toupper(age_grp),reas1=toupper(reas1))
+
+##################################################################
+
+#amend variables as necessary in order for matchi_numberng to work
+
+labeddaysdatasheetminusstandard <- left_join(datafile20,labedstemplate,
+                                             by = c(("LocalAuthorityArea" = "LocalAuthorityArea"), ("age_grp"="age_grp"),("reas1"="reas1")))
+
+arrange(labeddaysdatasheetminusstandard,LocalAuthorityArea,age_grp,reas1) # arrange data in order for tables
+
+#write_sav(labeddaysdatasheetminusstandard,paste0(filepath,"la bed days data sheet minus standard.sav")) # save out file
+labeddaysdatasheetminusstandard<-labeddaysdatasheetminusstandard%>% mutate(LocalAuthorityArea=toupper(LocalAuthorityArea),age_grp=toupper(age_grp),reas1=toupper(reas1))
+
+#Add in total for Standard filter on any row that isn't all
+datafile32a<-filter(labeddaysdatasheetminusstandard,reas1!="ALL") 
+#Rename reason_grp_high_level as Standard where there isn't a Code 9
+datafile32<-datafile32a%>% mutate(reas1=
+                                    if_else(reas1%in%c("HEALTH AND SOCIAL CARE REASONS","PATIENT/CARER/FAMILY-RELATED REASONS"),"STANDARD","CODE 9"))
+table(datafile32$reas1)
+
+#select standard only.
+datafile33<-filter(datafile32,reas1=="STANDARD")
+# aggregate LA Standard delays
+
+lastandard<- datafile33 %>% 
+  group_by(LocalAuthorityArea,age_grp,reas1) %>% 
+  summarise(OBDs=sum(OBDs,na.rm=TRUE)) %>% 
+  ungroup()
+
+#datafile24<-datafile24%>%rename(OBDs2=obds_in_month)
+
+lastandardandothers<- bind_rows(lastandard, labeddaysdatasheetminusstandard)
+lastandardandothers<-lastandardandothers%>% mutate(LocalAuthorityArea=toupper(LocalAuthorityArea),age_grp=toupper(age_grp),reas1=toupper(reas1))
+
+
+arrange(lastandardandothers,LocalAuthorityArea,age_grp,reas1) # issue here is that the rows with zeros don't appear so have to match to output file
+
+#datafile26<-read.csv(paste0(filepath2,"hb_template.csv"))
+#bhbbedstepmplate<-bhbbedstepmplate%>% mutate(nhs_board=toupper(nhs_board),age_grp=toupper(age_grp),reason_grp_high_level=toupper(reason_grp_high_level))
+
+#match files
+
+
+laallvariations <- left_join(labedstemplate,lastandardandothers,
+                             by = c(("LocalAuthorityArea" = "LocalAuthorityArea"), ("age_grp"="age_grp"),("reas1"="reas1")))
+
+#replace obds in correct column for all reason groups
+laallvariations<-laallvariations %>% mutate(OBDs=
+                                              if_else(is.na(OBDs),0L,OBDs))
+
+
+
+laallvariations<-arrange(laallvariations,laallvariations$LocalAuthorityArea,laallvariations$age_grp,laallvariations$reas1)
```

Comment:
See "Pipes" section in general feedback. I've also left comments above about the arguments required for `left_join()`. 
```suggestion
laallvariations <- left_join(labedstemplate, lastandardandothers, 
    by = c("LocalAuthorityArea", "age_grp", "reas1")) %>%
    # replace obds in correct column for all reason groups
    mutate(OBDs = if_else(is.na(OBDs), 0L, OBDs)) %>%
    arrange(LocalAuthorityArea, age_grp, reas1)
```